### PR TITLE
Fix: Backport Add Persistence for DataplaneSelector.

### PR DESCRIPTION
**Pull request by bmg13**
## WHAT

This PR backports the added DataPlane instance selector logic to the v0.7.4.

## WHY

To avoid the dataplane registry only in one ControlPlane in memory, this change enables the storage of DataPlane's availability in the bd. So, there should be enabled SQL persistence for data-plane-selector in backported 0.7.4 version.

## FURTHER NOTES

Logic here is partially extracted from https://github.com/eclipse-tractusx/tractusx-edc/pull/1475 PR

Closes https://github.com/eclipse-tractusx/tractusx-edc/issues/1497

### CODE REVIEW

**Comment by bmg13** [comment]
The data-plane-self-registration was added due to the need of including the dataplane unregister [here](https://github.com/eclipse-edc/Connector/pull/4249) and this change is not including in the 0.7.2 EDC version. This change was done in https://github.com/eclipse-tractusx/tractusx-edc/pull/1498/commits/5c58554b23b39c574ad15007bb00cca7de24d0ab.

**Comment by bmg13** [comment]
@paullatzelsperger @wolf4ood can you please review? 🙇

File: edc-extensions/dataplane/dataplane-self-registration/src/main/resources/META-INF/services/org.eclipse.edc.spi.system.ServiceExtension [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#################################################################################
+#  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```
Comment (wolf4ood) [inline edc-extensions/dataplane/dataplane-self-registration/src/main/resources/META-INF/services/org.eclipse.edc.spi.system.ServiceExtension:2]:
```suggestion
#  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
```

**Review by wolf4ood — APPROVED** [review]

File: edc-extensions/dataplane/dataplane-self-registration/src/main/resources/META-INF/services/org.eclipse.edc.spi.system.ServiceExtension (hunk 1 above)
Comment (bmg13) [reply edc-extensions/dataplane/dataplane-self-registration/src/main/resources/META-INF/services/org.eclipse.edc.spi.system.ServiceExtension]: Nice catch! Changed in https://github.com/eclipse-tractusx/tractusx-edc/pull/1498/commits/040a5029507945d6671eeef243ad37647183e9ee

**Comment by rafaelmag110** [comment]
@wolf4ood can you rerun the failing job?

**Comment by bmg13** [comment]
@wolf4ood can this be merged, please?

**Comment by wolf4ood** [comment]
@bmg13 done

**Pull request merged**